app-editors/zed/zed-0.150.4.ebuild

```diff
@@ -0,0 +1,1144 @@
+	[font-kit]='https://github.com/zed-industries/font-kit;40391b7c0041d8a8572af2afa3de32ae088f0120;font-kit-%commit%'
+	[lsp-types]='https://github.com/zed-industries/lsp-types;72357d6f6d212bdffba3b5ef4b31d8ca856058e7;lsp-types-%commit%'
+	[nvim-rs]='https://github.com/KillTheMule/nvim-rs;69500bae73b8b3f02a05b7bee621a0d0e633da6c;nvim-rs-%commit%'
+	[tree-sitter-gomod]='https://github.com/camdencheek/tree-sitter-go-mod;1f55029bacd0a6a11f6eb894c4312d429dcf735c;tree-sitter-go-mod-%commit%'
+	[tree-sitter-gowork]='https://github.com/d1y/tree-sitter-go-work;dcbabff454703c3a4bc98a23cf8778d4be46fd22;tree-sitter-go-work-%commit%'
+	[tree-sitter-heex]='https://github.com/phoenixframework/tree-sitter-heex;6dd0303acf7138dd2b9b432a229e16539581c701;tree-sitter-heex-%commit%'
+	[tree-sitter-language]='https://github.com/tree-sitter/tree-sitter;7f4a57817d58a2f134fe863674acad6bbf007228;tree-sitter-%commit%/lib/language'
+	[tree-sitter-md]='https://github.com/zed-industries/tree-sitter-markdown;e3855e37f8f2c71aa7513c18a9c95fb7461b1b10;tree-sitter-markdown-%commit%'
+	[tree-sitter]='https://github.com/tree-sitter/tree-sitter;7f4a57817d58a2f134fe863674acad6bbf007228;tree-sitter-%commit%/lib'
+	[xim-ctext]='https://github.com/npmania/xim-rs;27132caffc5b9bc9c432ca4afad184ab6e7c16af;xim-rs-%commit%/xim-ctext'
+	[xim-parser]='https://github.com/npmania/xim-rs;27132caffc5b9bc9c432ca4afad184ab6e7c16af;xim-rs-%commit%/xim-parser'
+	[xim]='https://github.com/npmania/xim-rs;27132caffc5b9bc9c432ca4afad184ab6e7c16af;xim-rs-%commit%'
+	[xkbcommon]='https://github.com/ConradIrwin/xkbcommon-rs;fcbb4612185cc129ceeff51d22f7fb51810a03b2;xkbcommon-rs-%commit%'
+)
+
+inherit cargo desktop flag-o-matic toolchain-funcs xdg
+
+DESCRIPTION="The fast, collaborative code editor."
+HOMEPAGE="https://zed.dev https://github.com/zed-industries/zed"
+SRC_URI="
+	https://github.com/zed-industries/zed/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz
+	${CARGO_CRATE_URIS}"
+
+LICENSE="0BSD AGPL-3+ Apache-2.0 Apache-2.0-with-LLVM-exceptions BSD BSD-2 Boost-1.0 CC0-1.0 GPL-3+ ISC LGPL-3 MIT MPL-2.0 Unicode-DFS-2016 Unlicense ZLIB"
+SLOT="0"
+KEYWORDS="~amd64"
+
+DEPEND="
+	app-arch/zstd:=
+	dev-db/sqlite:3
+	dev-libs/libgit2:=
+	dev-libs/mimalloc
+	dev-libs/openssl:0/3
+	dev-libs/wayland
+	dev-libs/wayland-protocols
+	dev-util/wayland-scanner
+	dev-util/vulkan-tools
+	media-libs/alsa-lib
+	media-libs/fontconfig
+	media-libs/vulkan-loader
+	net-misc/curl
+	x11-libs/libxcb
+	x11-libs/libxkbcommon
+"
+RDEPEND="${DEPEND}"
+BDEPEND="
+	>=virtual/rust-1.80.0
+	dev-util/vulkan-headers
+	sys-devel/gettext
+	sys-devel/mold
+"
+
+pkg_setup() {
+	if tc-is-gcc; then
+		export CARGO_PROFILE_RELEASE_LTO="true"
+	elif tc-is-clang; then
+		export CARGO_PROFILE_RELEASE_LTO="thin"
+	fi
+	strip-unsupported-flags
+	# flags from upstream
+	export RUSTFLAGS="${RUSTFLAGS} -C symbol-mangling-version=v0 --cfg tokio_unstable -C link-arg=-fuse-ld=mold -C link-args=-Wl,--disable-new-dtags,-rpath,\$ORIGIN/../lib"
```

Comment:
We now have **tc-ld-is-mold** function that can be used to detect if mold is used. I'd otherwise try forcing lld instead.